File: Client/src/StompMessage.cpp
//
// Created by atzmi@wincs.cs.bgu.ac.il on 12/01/2020.
//

#include "StompMessage.h"

StompMessage::StompMessage() : type(""), body(""), headersMap() {}

StompMessage::StompMessage(const StompMessage& other) : type(other.type), body(other.body), headersMap() {
    for(const auto& header : other.getHeadersMap()) {
        headersMap[header.first] = header.second;
    }
}

StompMessage* StompMessage::clone() {
    return new StompMessage(*this);
}

void StompMessage::setType(const std::string& type) {
    this->type = type;
}

void StompMessage::setHeader(const std::string& k, const std::string& v) {
    headersMap[k] = v;
}

void StompMessage::setBody(const std::string& body) {
    this->body = body;
}

const std::string& StompMessage::getBody() const {
    return body;
}

const std::string& StompMessage::getType() const {
    return type;
}

const std::string& StompMessage::getHeader(const std::string& key) {
    std::string output = "";
    if(headersMap.find(key) != headersMap.end())
        output = headersMap[key];
    return headersMap[key];
}

const std::unordered_map<std::string, std::string>& StompMessage::getHeadersMap() const {
    return headersMap;
}

StompMessage::~StompMessage() {
    headersMap.clear();
}

File: Server/src/main/java/bgu/spl/net/srv/StompMessage.java
package bgu.spl.net.srv;

import java.util.HashMap;
import java.util.Map;

public class StompMessage {

    private String type;
    private Map<String, String> headersMap;
    private String body;

    public StompMessage() {
        type = "";
        headersMap = new HashMap<>();
        body = "";
    }

    public StompMessage(StompMessage other) {
        this.type = other.type;
        this.headersMap = new HashMap<>();
        for (Map.Entry<String, String> p : other.headersMap.entrySet()) {
            this.setHeader(p.getKey(), p.getValue());
        }
        this.body = other.body;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setHeader(String k, String v) {
        this.headersMap.put(k, v);
    }

    public void setBody(String body) {
        this.body = body;
    }

    public String getBody() { return body; }

    public String getType() {
        return type;
    }

    public String getHeader(String key) {
        return headersMap.get(key);
    }

    public Map<String, String> getHeadersMap() {
        return headersMap;
    }

    @Override
    public String toString() {
        StringBuilder output = new StringBuilder();
        output.append(type).append("\n");
        for(Map.Entry<String, String> header : headersMap.entrySet()) {
            output.append(header.getKey()).append(":").append(header.getValue()).append("\n");
        }
        output.append("\n");
        output.append(body).append("\n");
        output.append("\u0000");
        return output.toString();
    }
}
